src/track/track.h

```diff
@@ -561,6 +566,8 @@ class Track : public QObject {
 
     // Storage for the track's beats
     mixxx::BeatsPointer m_pBeats;
+    QList<mixxx::BeatsPointer> m_pBeatsUndoStack;
```

Comment:
Here we may use QStack with push() and pop()

Reply:
Good to learn about QStack. So the benefit is to use push() and pop() instead of append() and takeLast()/removeLast(). Dunno if that is worth the extra include, but that's tiny, so I'll use it.